lib/smart_proxy_pulp_plugin/pulpcore_client.rb

```diff
@@ -27,9 +27,13 @@ def self.logger
     end
 
     def self.http
+      ssl_ca = ::PulpProxy::PulpcorePlugin.settings.ssl_ca ||
+               ::Proxy::SETTINGS.foreman_ssl_ca ||
+               ::Proxy::SETTINGS.ssl_ca_file
```

Comment:
Why are there three possibilities?

Reply:
My trail of thought was:
1. in most cases pulp will use the same CA as Foreman
2. fuck, foreman has TWO CAs
3. maybe someone was stupid enough to use a third CA for pulp

Reply:
Oh right, so you are basically saying:

1. If a CA is configured in the Plugin use that
2. If a foreman client CA was configured on the smart-proxy use that
3. If only the server CA was configured (meaning it's intended for server and client) then use that

Reply:
Correct!

Reply:
> in most cases pulp will use the same CA as Foreman

Technically, `ssl_ca_file`, which is what the Smart Proxy uses so this is "Smart Proxy and Pulp use the same CA"

> fuck, foreman has TWO CAs

`foreman_ssl_ca` is the CA that Foreman uses, so this covers the case where Foreman and Smart Proxy use different certs. For example, you use Let's Encrypt on Foreman and Puppet CA for Smart Proxy. I think it's reasonable to expect Pulp and Foreman to use the same CA since clients talk to both while Smart Proxy is IMHO still more of an internal service (though with global registration this really isn't true anymore).

> maybe someone was stupid enough to use a third CA for pulp

I think it is sensible to at least prepare the config options. Better to have it and not need it, than to need it and not have it. It's minimal code and makes it more flexible if needed.

Note https://github.com/theforeman/smart-proxy/blob/60aefda952b7c89f995c6ce872de63f3dc52ca08/lib/proxy/request.rb#L71 already does the same for option 1 and 2. This just adds option 3, which (as I said before) makes sense to me.

A general remark that I wouldn't mind if the Smart Proxy provided some internal APIs to make this easier. You can make the observation that smart-proxy and its various plugins have multiple abstractions over HTTP.